pkgs/games/betterspades/default.nix

```
@@ -0,0 +1,286 @@
+    openal
+    libdeflate
+    libX11
+    libXau
+    libXdmcp
+    libSM
+    libXext
+  ];
+
+  cmakeFlags = [
+    "-DFETCHCONTENT_SOURCE_DIR_VXL=${srcs.libvxl}"
+    "-DFETCHCONTENT_SOURCE_DIR_CGLM=${cglm.src}"
+  ];
+
+  # all .c and .h files in repo have CRLF line endings. This makes patching difficult.
+  prePatch = ''
+    find . -name '*.c' -exec dos2unix {} +
+    find . -name '*.h' -exec dos2unix {} +
+  '';
+
+  patches = [
+    ./001-dont-install-OpenAL32.dll.patch
+    ./002-fix-macos-build.patch
+  ];
+
+  postUnpack = ''
+    cp ${srcs."ini.c"} source/src/ini.c
+    cp ${srcs."ini.h"} source/src/ini.h
+
+    mkdir -p source/src/lodepng
+    cp ${srcs."lodepng.cpp"} source/src/lodepng/lodepng.c
+    cp ${srcs."lodepng.h"} source/src/lodepng/lodepng.h
+
+    cp ${srcs."dr_wav.c"} source/src/dr_wav.c
+
+    cp ${srcs."stb_truetype.h"} source/src/stb_truetype.h
+
+    cp ${srcs."parson.c"} source/src/parson.c
+    cp ${srcs."parson.h"} source/src/parson.h
+
+    cp ${srcs."http.h"} source/src/http.h
+
+    cp ${srcs."log.c"} source/src/log.c
+    cp ${srcs."log.h"} source/src/log.h
+
+    cp ${srcs."hashtable.c"} source/src/hashtable.c
+    cp ${srcs."hashtable.h"} source/src/hashtable.h
+
+    cp ${srcs."libdeflate.h"} source/deps/libdeflate.h
+
+    cp ${bsresources} source/bsresources.zip
+
+    cp ${srcs."microui.c"} source/src/microui.c
+    cp ${srcs."microui.h"} source/src/microui.h
+  '';
+
+  postInstall = ''
+      cd $out
+      rm -rf -- cache/ include/ lib/ logs/ screenshots/ config.ini
+      mkdir -p -- bin
+      mv -- client bin/.betterspades-wrapped
+      mkdir -p -- share/data
+      mv -- fonts/ kv6/ png/ wav/ share/data
+      mkdir -p -- share/icons/hicolor/256x256/apps
+      cp -- "${icon}" share/icons/hicolor/256x256/apps/betterspades.png
+      cp -- "${src}/resources/config.ini" share/data/config.ini
+      cp -- ${./wrapper.sh} bin/betterspades
+      chmod +x bin/betterspades
+  '';
+
+  desktopItems =
+    let
+      desktop = {
+        comment = meta.description;
+        genericName = "Sandbox building and FPS videogame";
+        icon = pname;
+        categories = [ "Game" "ActionGame" ];
+        keywords = [ "game" "fps" "sandbox" "voxel" "aos" ];
+        prefersNonDefaultGPU = true;
+      };
+      desktop_name = "BetterSpades";
+    in
+    [
+      (makeDesktopItem ({
+        name = pname;
+        desktopName = desktop_name;
+        exec = "betterspades";
+      } // desktop))
+
+      (makeDesktopItem ({
+        name = "${pname}.aosProtocolHandler";
+        desktopName = "${desktop_name} (protocol handler)";
+        exec = "betterspades -%U";
+        mimeTypes = [ "x-scheme-handler/aos" ];
+        noDisplay = true;
+      } // desktop))
+    ];
+
+  passthru.updateScript = ./update.sh;
+
+  meta = with lib; {
+    description = "Ace of Spades 0.75 client targeted at low end systems";
+    homepage = "https://github.com/xtreme8000/BetterSpades/";
+    license = licenses.gpl3;
```

gpl3 license is depreacted, please use gpl3Only or gpl3Plus

---

If there is no mention of plus then it is gpl3Only

```suggestion
    license = licenses.gpl3Only;
```